### PR TITLE
Feat drawing tool integration

### PR DESCRIPTION
**Pull request by ahmadtaimoor-deriv**

### CODE REVIEW

File: chart_app/lib/src/models/indicators.dart [hunk 1]
```diff
@@ -320,21 +318,21 @@ class IndicatorsModel {
                   item.jawSeries!.entries,
                   epoch,
                   pipSize,
-                  offset: item.alligatorOptions.jawOffset,
+                  // offset: item.alligatorOptions.jawOffset,
```
Comment (balakrishna-deriv) [inline chart_app/lib/src/models/indicators.dart:321]: Why are we removing the offset?

File: chart_app/lib/src/models/indicators.dart [hunk 2]
```diff
@@ -320,21 +318,21 @@ class IndicatorsModel {
...
                 ),
               if (item.teethSeries != null)
                 _getQuote(
                   item.teethSeries!.entries,
                   epoch,
                   pipSize,
-                  offset: item.alligatorOptions.teethOffset,
+                  // offset: item.alligatorOptions.teethOffset,
```
Comment (balakrishna-deriv) [inline chart_app/lib/src/models/indicators.dart:328]: here too

File: src/store/ChartAdapterStore.ts [hunk 3]
```diff
@@ -192,6 +191,30 @@ export default class ChartAdapterStore {
...
+                onUpdate: () => {
+                    this.mainStore.drawTools.onUpdate();
+                },
+                onRemove: () => {
+                    // this.mainStore.drawTools.deleteGroupTool(index);
+                },
+                onEdit: () => {
+                    // console.log('From the On Edit', index);
+                    // this.mainStore.studies.editStudyByIndex(index);
+                },
```
Comment (balakrishna-deriv) [inline src/store/ChartAdapterStore.ts:207]: Let's remove these if they are not needed.

File: chart_app/lib/src/models/indicators.dart (hunk 2 above)
Comment (ahmadtaimoor-deriv) [reply chart_app/lib/src/models/indicators.dart]: It was commented while merging restoreDrawing branch into this branch and there were some error conflicts . It is added back now , Thanks

File: chart_app/lib/src/models/indicators.dart (hunk 1 above)
Comment (ahmadtaimoor-deriv) [reply chart_app/lib/src/models/indicators.dart]: It was commented while merging restoreDrawing branch into this branch and there were some error conflicts . It is added back now , Thanks

File: src/store/ChartAdapterStore.ts (hunk 3 above)
Comment (ahmadtaimoor-deriv) [reply src/store/ChartAdapterStore.ts]: Removed now , thanks

File: src/store/ChartState.ts [hunk 4]
```diff
@@ -81,9 +81,9 @@ class ChartState {
     get loader() {
         return this.mainStore.loader;
     }
-    get drawTools() {
-        return this.mainStore.drawTools;
-    }
+    // get drawTools() {
+    //     return this.mainStore.drawTools;
+    // }
```
Comment (balakrishna-deriv) [inline src/store/ChartState.ts:86]: Please remove this if it's not needed

Comment (ahmadtaimoor-deriv) [reply src/store/ChartState.ts]: Removed , Thanks.

**Review by balakrishna-deriv — APPROVED** [review]

File: chart_app/lib/main.dart [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import 'dart:async';
```
Comment (balakrishna-deriv) [inline chart_app/lib/main.dart:1]: We don't need this import now, right?

File: chart_app/lib/src/models/chart_config.dart [hunk 6]
```diff
@@ -136,6 +149,13 @@ class ChartConfigModel extends ChangeNotifier {
     notifyListeners();
   }
 
+  /// To set the time interval
+  // ignore: avoid_positional_boolean_parameters
+  void setTimeInterval(bool showInterval) {
```
Comment (balakrishna-deriv) [inline chart_app/lib/src/models/chart_config.dart:154]:
```suggestion
  void toggleTimeIntervalVisibility(bool showInterval) {
```

**Pull request merged**